charts/shoot-core/components/charts/apiserver-proxy/templates/apiserver-proxy-pod-mutatingwebhook.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: {{ include "webhookadmissionregistration" . }}
+kind: MutatingWebhookConfiguration
+metadata:
+  name: apiserver-proxy.networking.gardener.cloud
+  labels:
+    gardener.cloud/role: system-component
+    origin: gardener
+webhooks:
+- admissionReviewVersions:
+  - v1beta1
+  clientConfig:
+    caBundle: {{ .Values.webhook.caBundle }}
+    url: https://127.0.0.1:9443/webhook/pod-apiserver-env
+  failurePolicy: Ignore
+  matchPolicy: Exact
+  name: apiserver-proxy.networking.gardener.cloud
+  namespaceSelector:
+    matchExpressions:
+    - key: apiserver-proxy.networking.gardener.cloud/inject
+      operator: NotIn
+      values:
+      - disable
+  objectSelector:
+    matchExpressions:
+    - key: apiserver-proxy.networking.gardener.cloud/inject
+      operator: NotIn
+      values:
+      - disable
```

Comment:
I guess this breaks for shoots with an old Kubernetes version, right? Gardener is still capable of managing Kubernetes 1.10, so this whole manifest should be adapted to cater with all versions.

Reply:
How so? This resource is created by Gardener resource manager. It'll send this field, but it'll be ignored by the API server.

Reply:
Is it? OK, if so, fine. Just wanted to point that out :)

Reply:
```
Request Body: {"THIS_FIELD_DOES_NOT_EXIST":"bar","apiVersion":"v1","kind":"ConfigMap","metadata":{"name":"foo","namespace":"default"}}
curl -k -v -XPOST  -H "Content-Type: application/json" -H "Accept: application/json" -H "User-Agent: kubectl/v1.18.6 (darwin/amd64) kubernetes/dff82dc" 'https://kubernetes.docker.internal:6443/api/v1/namespaces/default/configmaps'
POST https://kubernetes.docker.internal:6443/api/v1/namespaces/default/configmaps 201 Created in 8 milliseconds
Response Headers:
    Content-Length: 258
    Date: Mon, 02 Nov 2020 07:29:04 GMT
    Cache-Control: no-cache, private
    Content-Type: application/json
Response Body: {"kind":"ConfigMap","apiVersion":"v1","metadata":{"name":"foo","namespace":"default","selfLink":"/api/v1/namespaces/default/configmaps/foo","uid":"eea2f54b-6ac8-4e0b-b88c-05b5740ddce4","resourceVersion":"1887256","creationTimestamp":"2020-11-02T07:29:04Z"}}
configmap/foo created
```

The client is sending some `THIS_FIELD_DOES_NOT_EXIST` field which is not stored by the API server. The validation is only on the client side via OpenAPI.